src/index.js

```diff
@@ -194,34 +195,39 @@ class Dropzone extends React.Component {
       const acceptedFiles = []
       const rejectedFiles = []
 
-      fileList.forEach(file => {
-        if (!disablePreview) {
-          try {
-            file.preview = window.URL.createObjectURL(file) // eslint-disable-line no-param-reassign
-          } catch (err) {
-            if (process.env.NODE_ENV !== 'production') {
-              console.error('Failed to generate preview for file', file, err) // eslint-disable-line no-console
+      if (!isFileDataTransferItems(evt)) {
```

Comment:
Actually, all you need here is:
```js
if (
  !isFileList(fileList) ||
  (isIeOrEdge() && evt.dataTransfer && !isFileList(Array.from(evt.dataTransfer.items)))
) {
  return
}
```

Everything else in the PR can be removed. And you need to fix one of the preview test cases.

I'd also add some tests for this use case to ensure everything works as expected when IE/Edge is used.

I tested this in Edge 17/16 on SauceLabs.